scalyr_agent/configuration.py

```diff
@@ -250,6 +252,14 @@ def parse(self):
 
             self.__monitor_configs = list(self.__config.get_json_array("monitors"))
 
+            # Set json library based on the config value. If "auto" is provided this means we use
+            # default behavior which is try to use ujson and if that's not available fall back to
+            # stdlib json
+            json_library = self.__config.get_string("json_library")
```

Comment:
Setting this change within the `parse` method isn't quite correct I think with our current model.  Remember, every 30 seconds, we are parsing the configuration state from disk to see if it has changed.  If it is different, then we go ahead and apply those changes by stopping the current monitors and copying manager and starting new ones with the new configuration.  Given that model, I'd be surprised as a developer that the `parse` method has a side effect like this.

What we probably need to do is have a `apply_config` method in something like `agent_main.py` (where the new configuration state is being read) to apply any changes like this to the global state.

Reply:
Yeah, I agree about ``parse()`` side affect.

For now, I will move that functionality into ``__create_worker_thread`` and in the future, we can refactor it a bit more and have additional apply config or similar method.